pkg/client/client.go

```diff
@@ -227,12 +226,11 @@ func (o *OptimizelyClient) GetFeatureVariableJSON(featureKey, variableKey string
 		return value, err
 	}
 
-	var convertedValue map[string]interface{}
-	err = json.Unmarshal([]byte(val), &convertedValue)
+	value, err = optimizelyjson.NewOptimizelyJSONfromString(val)
 	if err != nil || valueType != entities.JSON {
-		return value, fmt.Errorf("variable value for key %s is invalid or wrong type", variableKey)
+		return nil, fmt.Errorf("variable value for key %s is invalid or wrong type", variableKey)
 	}
-	value = optimizelyjson.NewOptimizelyJSONfromMap(convertedValue)
+
```

Comment:
Perhaps be:
```go
if valueType != entities.JSON {
	return nil, fmt.Errorf("variable value for key %s is invalid or wrong type", variableKey)
}

return optimizelyjson.NewOptimizelyJSONfromString(val)
```

Reply:
since we have the same message for all variables ("invalid or wrong type") then the variable validation has to happen first.